platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentV3DialcodeTest.java

```diff
@@ -0,0 +1,479 @@
+        List<String> reservedDialcodesNew = response.jsonPath().get("result.reservedDialcodes");
+        assertEquals(20, reservedDialcodesNew.size());
+        assertTrue(reservedDialcodesNew.containsAll(reservedDialcodes));
+
+    }
+
+    @Test
+    public void reserveDialcodesTwiceWithHigherCountSecondTime() {
+        setURI();
+        String createTextBookRequest = this.createTextBookRequest.replace(IDENTIFIER, "" + generateRandomInt(0, 99999));
+        String[] result = this.contentV3API.createAndGetResult(createTextBookRequest, getRequestSpecification(contentType, validuserId, APIToken, channelId, appId), get200ResponseSpec());
+        String identifier = result[0];
+
+        String reserveDialcodesRequestBody = this.reserveDialcodesRequestBody.replace(COUNT, "" + 20);
+        Response response = this.contentV3API.reserveDialcode(identifier, reserveDialcodesRequestBody, getRequestSpecification(contentType, validuserId, APIToken, channelId, appId), get200ResponseSpec());
+        List<String> reservedDialcodes = response.jsonPath().get("result.reservedDialcodes");
+        assertEquals(20, reservedDialcodes.size());
+
+        reserveDialcodesRequestBody = this.reserveDialcodesRequestBody.replace(COUNT, "" + 25);
+        response = this.contentV3API.reserveDialcode(identifier, reserveDialcodesRequestBody, getRequestSpecification(contentType, validuserId, APIToken, channelId, appId), get200ResponseSpec());
+        List<String> reservedDialcodesNew = response.jsonPath().get("result.reservedDialcodes");
+        assertEquals(25, reservedDialcodesNew.size());
+        assertTrue(reservedDialcodesNew.containsAll(reservedDialcodes));
+
+    }
+
+    @Test
+    public void reserveDialcodesTwiceWithLowerCountSecondTime() {
+        setURI();
+        String createTextBookRequest = this.createTextBookRequest.replace(IDENTIFIER, "" + generateRandomInt(0, 99999));
+        String[] result = this.contentV3API.createAndGetResult(createTextBookRequest, getRequestSpecification(contentType, validuserId, APIToken, channelId, appId), get200ResponseSpec());
+        String identifier = result[0];
+
+        String reserveDialcodesRequestBody = this.reserveDialcodesRequestBody.replace(COUNT, "" + 20);
+        Response response = this.contentV3API.reserveDialcode(identifier, reserveDialcodesRequestBody, getRequestSpecification(contentType, validuserId, APIToken, channelId, appId), get200ResponseSpec());
+        List<String> reservedDialcodes = response.jsonPath().get("result.reservedDialcodes");
+        assertEquals(20, reservedDialcodes.size());
+
+        reserveDialcodesRequestBody = this.reserveDialcodesRequestBody.replace(COUNT, "" + 15);
+        response = this.contentV3API.reserveDialcode(identifier, reserveDialcodesRequestBody, getRequestSpecification(contentType, validuserId, APIToken, channelId, appId), get200ResponseSpec());
```

Comment:
Second time API call should return client error message (It should not be 200).

Reply:
Refactored.